contracts/strategies/VariableRewardsStrategy.sol

```diff
@@ -0,0 +1,304 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.7.3;
+
+import "../YakStrategyV2.sol";
+import "../interfaces/IPair.sol";
+import "../lib/DexLibrary.sol";
+import "../lib/SafeERC20.sol";
+
+/**
+ * @notice VariableRewardsStrategy
+ */
+abstract contract VariableRewardsStrategy is YakStrategyV2 {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    IWAVAX internal constant WAVAX = IWAVAX(0xB31f66AA3C1e785363F0875A1B74E27b85FD66c7);
+
+    struct Reward {
+        address reward;
+        uint256 amount;
+    }
+
+    struct RewardSwapPairs {
+        address reward;
+        address swapPair;
+    }
+
+    struct StrategySettings {
+        uint256 minTokensToReinvest;
+        uint256 adminFeeBips;
+        uint256 devFeeBips;
+        uint256 reinvestRewardBips;
+    }
+
+    // reward -> swapPair
+    mapping(address => address) public rewardSwapPairs;
+    address[] public supportedRewards;
+    uint256 public rewardCount = 1;
+
+    constructor(
+        string memory _name,
+        address _depositToken,
+        RewardSwapPairs[] memory _rewardSwapPairs,
+        address _timelock,
+        StrategySettings memory _strategySettings
+    ) Ownable() {
+        name = _name;
+        depositToken = IERC20(_depositToken);
+        rewardToken = IERC20(address(WAVAX));
+        devAddr = 0x2D580F9CF2fB2D09BC411532988F2aFdA4E7BefF;
+
+        for (uint256 i = 0; i < _rewardSwapPairs.length; i++) {
+            _addReward(_rewardSwapPairs[i].reward, _rewardSwapPairs[i].swapPair);
+        }
+
+        updateMinTokensToReinvest(_strategySettings.minTokensToReinvest);
+        updateAdminFee(_strategySettings.adminFeeBips);
+        updateDevFee(_strategySettings.devFeeBips);
+        updateReinvestReward(_strategySettings.reinvestRewardBips);
+        updateDepositsEnabled(true);
+        transferOwnership(_timelock);
+        emit Reinvest(0, 0);
+    }
+
+    function addReward(address _rewardToken, address _swapPair) public onlyDev {
+        _addReward(_rewardToken, _swapPair);
+    }
+
+    function _addReward(address _rewardToken, address _swapPair) internal {
+        if (_rewardToken != address(rewardToken)) {
+            require(
+                DexLibrary.checkSwapPairCompatibility(IPair(_swapPair), _rewardToken, address(rewardToken)),
+                "VariableRewardsStrategy::Swap pair does not contain reward token"
+            );
+        }
+        rewardSwapPairs[_rewardToken] = _swapPair;
+        supportedRewards[rewardCount] = _rewardToken;
+        rewardCount = rewardCount.add(1);
+    }
+
+    function removeReward(address _rewardToken) public onlyDev {
```

Comment:
This should also cause an update in the `supportedRewards` array

Reply:
fixed